go/vt/sidecardb/sidecardb.go

```diff
@@ -342,6 +351,22 @@ func (si *schemaInit) setCurrentDatabase(dbName string) error {
 	return err
 }
 
+func (si *schemaInit) collation() (collations.ID, error) {
+	rs, err := si.exec(si.ctx, sidecarCollationQuery, 2, false)
+	if err != nil {
+		log.Error(err)
+		return collations.Unknown, err
+	}
+
+	switch len(rs.Rows) {
```

Comment:
I know this is a copy+paste from elsewhere, because I recognize the pattern. I think at some later stage we should cleanup this kind of code. A `select @@global.collation_server` will either fail, or return exactly one row. There is no need to check in the code what happens if less or more different rows are returned. It's only confusing and adding error prone logic.

Reply:
@shlomi-noach It actually helped me here since with the fake db I had to add stuff since things were missing since no mock queries were set up. So it help uncover that in an easier way than random panics.